<launch>
    <arg name="map_size_x_"/>
    <arg name="map_size_y_"/>
    <arg name="map_size_z_"/>
    <arg name="odometry_topic"/>
    <arg name="camera_pose_topic"/>
    <arg name="depth_topic"/>
    <arg name="cloud_topic"/>
    <arg name="cx"/>
    <arg name="cy"/>
    <arg name="fx"/>
    <arg name="fy"/>
    <arg name="point_num"/>
    <arg name="point0_x"/>
    <arg name="point0_y"/>
    <arg name="point0_z"/>
    <arg name="point1_x"/>
    <arg name="point1_y"/>
    <arg name="point1_z"/>
    <arg name="point2_x"/>
    <arg name="point2_y"/>
    <arg name="point2_z"/>
    <arg name="flight_type"/>
    <arg name="drone_id"/>
    <arg name="max_vel"/>
    <!-- main node -->
    <!-- <node pkg="primitive_planner" name="drone_$(arg drone_id)_primitive_planner_node" type="primitive_planner_node" output="screen" launch-prefix="valgrind"> -->
    <node pkg="primitive_planner" name="drone_$(arg drone_id)_primitive_planner_node" type="primitive_planner_node" output="screen">
        <remap from="~odom_world" to="/drone_$(arg drone_id)_$(arg odometry_topic)"/>
        <remap from="~mandatory_stop" to="/mandatory_stop_to_planner"/>
        <remap from="~planning/broadcast_primitive_send" to = "/broadcast_primitive_from_planner"/>
        <remap from="~planning/broadcast_primitive_recv" to = "/broadcast_primitive_to_planner"/>
        <remap from="~planning/selected_path_id" to = "/drone_$(arg drone_id)_planning/selected_path_id"/>
        <remap from="~planning/polynomial_traj" to = "/drone_$(arg drone_id)_planning/polynomial_traj"/>
        <remap from="~planning/yaw_cmd" to = "/drone_$(arg drone_id)_planning/yaw_cmd"/>
        <remap from="~planning/stop_command" to = "/drone_$(arg drone_id)_planning/stop_command"/>
        <remap from="~planning/heartbeat" to = "/drone_$(arg drone_id)_traj_server/heartbeat"/>
        <remap from="~planning/global_goal" to = "/drone_$(arg drone_id)_planning/global_goal"/>
        <remap from="~planning/select_path_end" to = "/drone_$(arg drone_id)_planning/select_path_end"/>
        <!-- rviz instruct goal -->
        <remap from="/goal" to = "/goal_with_id"/>
        <remap from="~plan_manage/odom" to = "/drone_$(arg drone_id)_$(arg odometry_topic)"/>
        <remap from="~plan_manage/cloud" to = "/drone_$(arg drone_id)_$(arg cloud_topic)"/>
        <remap from="position_cmd" to="drone_$(arg drone_id)_planning/pos_cmd"/>

        <!-- planning fsm -->
        <param name="fsm/thresh_replan_time" value="0.2" type="double"/>
        <param name="fsm/realworld_experiment" value="true"/>
        <param name="fsm/fail_safe" value="true"/>
        <param name="fsm/no_replan_thresh" value="0.5" type="double"/>
        <param name="fsm/flight_type" value="$(arg flight_type)" type="int"/>
        <param name="fsm/waypoint_num" value="$(arg point_num)" type="int"/>
        <param name="fsm/waypoint0_x" value="$(arg point0_x)" type="double"/>
        <param name="fsm/waypoint0_y" value="$(arg point0_y)" type="double"/>
        <param name="fsm/waypoint0_z" value="$(arg point0_z)" type="double"/>
        <param name="fsm/waypoint1_x" value="$(arg point1_x)" type="double"/>
        <param name="fsm/waypoint1_y" value="$(arg point1_y)" type="double"/>
        <param name="fsm/waypoint1_z" value="$(arg point1_z)" type="double"/>
        <param name="fsm/waypoint2_x" value="$(arg point2_x)" type="double"/>
        <param name="fsm/waypoint2_y" value="$(arg point2_y)" type="double"/>
        <param name="fsm/waypoint2_z" value="$(arg point2_z)" type="double"/>
        <!-- planner manager -->
        <param name="manager/primitiveFolder" value="$(find primitive_planner)/primitive_library" type="string" />
        <param name="manager/depthCloudStackNum" value="1" type="int"/>
        <param name="manager/boxX" value="6.0" type="double"/>
        <param name="manager/boxY" value="6.0" type="double"/>
        <param name="manager/boxZ" value="6.0" type="double"/>
        <param name="manager/arc_length" value="3.0" type="double"/>
        <param name="manager/voxelSize" value="0.1" type="double"/>
        <!-- <param name="manager/pathNum" value="109" type="int"/> -->
        <param name="manager/sampleSize" value="10000" type="int"/>
        <!-- <param name="manager/lamda_c" value="150.0" type="double"/> -->
        <param name="manager/lamda_l" value="200.0" type="double"/>
        <param name="manager/lamda_b" value="50.0" type="double"/>
        <param name="manager/drone_id" value="$(arg drone_id)"/>
        <param name="manager/map_size_x" value="$(arg map_size_x_)" />
        <param name="manager/map_size_y" value="$(arg map_size_y_)" />
        <param name="manager/map_size_z" value="$(arg map_size_z_)" />
        <param name="manager/max_vel" value="$(arg max_vel)"/>
        <!-- Drone radius r = 0.12 -->
        <param name="manager/swarm_clearence" value="0.24" type="double"/>
    </node>
</launch>
